datafusion/common/src/types/logical.rs

```diff
@@ -32,7 +34,7 @@ pub enum TypeSignature<'a> {
     /// The `name` should contain the same value as 'ARROW:extension:name'.
     Extension {
         name: &'a str,
-        parameters: &'a [TypeParameter<'a>],
+        parameters: Vec<TypeParameter<'a>>,
```

Comment:
I was wrangling with the lifetime 'a as it's both used for the parameter and the slice. We can change that once we clean-up the PR if we find a better solution.